packages/mgt-components/src/components/mgt-file/mgt-file.ts

```diff
@@ -428,25 +433,20 @@ export class MgtFile extends MgtTemplatedComponent {
     }
 
     const file = this.driveItem;
-    let fileTemplate;
+    let fileTemplate: TemplateResult;
 
     fileTemplate = this.renderTemplate('default', { file });
     if (!fileTemplate) {
       const fileDetailsTemplate: TemplateResult = this.renderDetails(file);
       const fileTypeIconTemplate: TemplateResult = this.renderFileTypeIcon();
 
       fileTemplate = html`
-        <div class="item">
+        <div tabindex="0" class="item">
```

Comment:
Should we really be setting a tab index on this element?
I ask as I can't see a click handler or other interactive behavior for this element.

Reply:
I added that for the keyboard navigation (focus & focus-within states) in mgt-file as well as keyboard navigation in the mgt-file-list. Also, now I'm thinking: in general, when you have a component like mgt-file and it's used in another component like mgt-file-list, what is the best place to have this _functionality_? `mgt-file-list` seems like a reasonable place to have it and in such a case, this will be _unnecessary_ here.

Reply:
This is another example of how the loosely coupled interaction mechanism that we have makes it hard to provide good a11y hooks.

In terms of adding focus markers for keyboard navigation, my understanding here is that we should only have a tabindex set on elements that are interactive, i.e. they have a click hander. In the case of list-based navigation IMO that should be added by the list wrapper rather than the item. From my experimentation lists like this typically use the arrow keys to navigate within them, and the tab key will only be used to select the first item in the list. 

Have a play with the document library here using the keyboard to see what I mean: https://microsoft.sharepoint.com/teams/MSGraph/Shared%20Documents/Forms/AllItems.aspx?RootFolder=%2Fteams%2FMSGraph%2FShared%20Documents%2FOrg%20Teams%2FDevX%2FSpecs%2C%20one%20pagers%20%26%20Design%20docs%2FGraph%20Explorer

Reply:
Now I understand this especially with the added example. I have updated the file list to use Arrow Up/Down to navigate the list and only use the tab to focus the file items.